src/software/ai/hl/stp/tactic/stop_fsm.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/stop_intent.h"
+
+struct StopFSM
+{
+    class idle_state;
+    class stop_state;
+
+    struct Update
+    {
+        Update(bool coast, const TacticUpdate& common) : coast(coast), common(common) {}
+        bool coast;
```

Comment:
we should pop `coast` in a `ControlParams` struct and use the macro here

Reply:
done